modules/tinymce/src/plugins/media/main/ts/ui/Dialog.ts

```diff
@@ -90,9 +91,20 @@ const handleError = (editor: Editor) => (error?: { msg: string }): void => {
 const getEditorData = (editor: Editor): MediaData => {
   const element = editor.selection.getNode();
   const snippet = isMediaElement(element) ? editor.serializer.serialize(element, { selection: true }) : '';
+  const data = HtmlToData.htmlToData(snippet, editor.schema);
+
+  const optDimensions = isEmbedIframe(data.type, data.source) ? Optional.some(editor.dom.getRect(element)) : Optional.none();
+  const dimensions = optDimensions.map((x) => {
+    return {
+      width: x.w.toString().replace(/px$/, ''),
+      height: x.h.toString().replace(/px$/, ''),
+    };
+  }).getOr({});
```

Comment:
Can do without Optionals:
```suggestion
  const getDimensionsOfElement = (): MediaDialogData['dimensions'] => {
    const rect = editor.dom.getRect(element);
    return {
      width: rect.w.toString().replace(/px$/, ''),
      height: rect.h.toString().replace(/px$/, ''),
    };
  };
  const dimensions = isEmbedIframe(data.type, data.source) ? getDimensionsOfElement() : {};
```

Reply:
Or even this:
```suggestion
  const getDimensionsOfElement = (): MediaDialogData['dimensions'] => {
    if (isEmbedIframe(data.type, data.source)) {
      const rect = editor.dom.getRect(element);
      return {
        width: rect.w.toString().replace(/px$/, ''),
        height: rect.h.toString().replace(/px$/, ''),
      };
    } else {
      return {};
    }
  };
  const dimensions = getDimensionsOfElement();
```